### PR TITLE
Feature/architecture

### PR DESCRIPTION
**Pull request by AElkhami**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -21,7 +22,16 @@ android {
     }
 
     buildTypes {
+        debug {
+            buildConfigField "String", "AUTH_KEY","\"Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJjZWVkZTk2ZDUxNTQ4NTA1ZTY3YTY4NjFhY2U3NjBmYyIsInN1YiI6IjY0YjY1MmUxMzc4MDYyMDBmZjNhNWM0NyIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.g3GD-e6kvUmXyPXG8EJ8lAYPBEDZQy4LQzrmn8rngBA\""
```
Comment (MohamedHashim) [inline app/build.gradle]: From a security perspective, We shouldn't reveal the API key like this as it's exposed to the public and anyone can use it.
Try to add this key in gradle.properties file and then add this file to .gitignore file to make it invisible in the version control.

File: app/src/main/java/com/example/aimovies/presentation/home/HomeScreen.kt [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        releaseDate = "01/03/2023"
+    ), MovieModel(
+        overview = "test",
+        title = "The Tomorrow War",
+        voteCount = 10,
+        voteAverage = 8.5,
+        posterPath = "https://assets-global.website-files.com/6009ec8cda7f305645c9d91b/6408f676b5811234c887ca62_top%20gun%20maverick-min.png",
+        releaseDate = "01/03/2023"
+    )
+)
```
Comment (MohamedHashim) [inline app/src/main/java/com/example/aimovies/presentation/home/HomeScreen.kt]: Why do we need to have static movies while getting them through API?

File: app/src/main/java/com/example/aimovies/presentation/home/HomeScreen.kt [hunk 3]
```diff
@@ -171,7 +146,7 @@ fun HomeScreenUi(
 @Preview(showBackground = true)
 fun HomeScreenPreview() {
     AIMoviesTheme {
-        HomeScreenUi(HomeUiModel(movies)) {
+        HomeScreenUi(HomeUiModel(emptyList())) {
```
Comment (MohamedHashim) [inline app/src/main/java/com/example/aimovies/presentation/home/HomeScreen.kt]: You can use HomeUiModel() without emptyList() as it has a default value of emptyList().

File: app/src/main/java/com/example/aimovies/domain/use_case/GetFavouriteMovies.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.example.aimovies.data.repository.favourite.FavouriteMovieRepository
+import example.moviedb.FavouriteEntitiy
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * Created by A.Elkhami on 25/07/2023.
+ */
+class GetFavouriteMovies(private val repository: FavouriteMovieRepository) {
+    operator fun invoke(): Flow<List<FavouriteEntitiy>> {
```
Comment (MohamedHashim) [inline app/src/main/java/com/example/aimovies/domain/use_case/GetFavouriteMovies.kt:11]: I think this function should be suspend function.

**Review by MohamedHashim — APPROVED** [review]
Good job!

File: app/src/main/java/com/example/aimovies/domain/use_case/GetFavouriteMovies.kt (hunk 4 above)
Comment (AElkhami) [reply app/src/main/java/com/example/aimovies/domain/use_case/GetFavouriteMovies.kt:11]: This function returns flow and doesn't call any other suspend function, so it doesn't need to be suspended as the flow will do anything related async within itself.

**Pull request merged**